sgxs-tools/src/sgx_detect/interpret.rs

```diff
@@ -354,3 +356,26 @@ mod serde {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum DSStatus {
+    Running,
+    NotRunning,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NodeAgentStatus{
+    pub current_ver: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum BuildType {
+    Generic,
+    EnclaveOSPreInstall,
+    EnclaveOSPostInstall,
+}
+
+lazy_static! {
+    pub static ref BUILD_FOR: Mutex<BuildType> = Mutex::new(BuildType::Generic);
```

Comment:
I think passing in `BuildType` into `fn detect` as well as `print` (add it beside the `verbose` argument) would be sufficient.